casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala

```diff
@@ -47,7 +47,7 @@ object InterpreterUtil {
       _                    <- Span[F].mark("before-unsafe-get-parents")
       parents              <- ProtoUtil.getParents[F](b)
       _                    <- Span[F].mark("before-compute-parents-post-state")
-      possiblePreStateHash <- computeParentsPostState[F](parents, dag, runtimeManager)
+      possiblePreStateHash <- computeParentsPostState[F](parents, dag, runtimeManager).attempt
```

Comment:
It's not clear to me why this is better. You're eliminating the `Either` return type from `computeParentsPostState`, but in exchange you have to re-throw the errors from `replayIntoMergeBlock`. Maybe it'd be worthwhile if you also eliminated the `Either` return type of `replayIntoMergeBlock`.

Reply:
I do exactly what you suggest in the latter commits :)